pufferlib/ocean/trash_pickup/trash_pickup.h

```diff
@@ -0,0 +1,596 @@
+        }
+
+        // Add shared trash data
+        for (int i = 0; i < env->num_trash; i++) {
+            if (env->trash_presence[i]) {
+                obs[obs_index++] = (float) (env->trash_positions[i][0]) * norm_factor;
+                obs[obs_index++] = (float) (env->trash_positions[i][1]) * norm_factor;
+                obs[obs_index++] = 1.0f; // Trash present
+            } else {
+                obs[obs_index++] = -1.0f; // Placeholder for x
+                obs[obs_index++] = -1.0f; // Placeholder for y
+                obs[obs_index++] = 0.0f; // Trash not present
+            }
+        }
+
+        // Add shared bin data
+        for (int i = 0; i < env->num_bins; i++) {
+            obs[obs_index++] = env->bin_positions[i][0] * norm_factor;
+            obs[obs_index++] = env->bin_positions[i][1] * norm_factor;
+        }
+    }
+}
+
+// Helper functions
+void place_random_items(CTrashPickupEnv* env, int count, int item_type) {
+    int placed = 0;
+    while (placed < count) 
+    {
+        int x = rand() % env->grid_size;
+        int y = rand() % env->grid_size;
+
+        if (env->grid[INDEX(env, x, y)].type == EMPTY) {
+            env->grid[INDEX(env, x, y)].type = item_type;
+
+            if (item_type == AGENT) {
+                env->grid[INDEX(env, x, y)].index = placed;
+                env->agent_positions[placed][0] = x;
+                env->agent_positions[placed][1] = y;
+            } else if (item_type == TRASH) {
+                env->grid[INDEX(env, x, y)].index = placed; // Set trash index
+                env->trash_positions[placed][0] = x;
+                env->trash_positions[placed][1] = y;
+                env->trash_presence[placed] = 1;
+            } else if (item_type == TRASH_BIN) {
+                env->grid[INDEX(env, x, y)].index = placed; // Set bin index
+                env->bin_positions[placed][0] = x;
+                env->bin_positions[placed][1] = y;
+            }
+
+            placed++;
+        }
+    }
+}
+
+void add_reward(CTrashPickupEnv* env, float reward){
+    // Reward all agents to promote cooperative behavior rather than competitive
+    for (int i = 0; i < env->num_agents; i++){
+         env->rewards[i] += reward;
+    }
+
+    env->total_episode_reward += reward;
+}
+
+void move_agent(CTrashPickupEnv* env, int agent_idx, int action) {
+    int x = env->agent_positions[agent_idx][0];
+    int y = env->agent_positions[agent_idx][1];
+    int carrying = env->agent_carrying[agent_idx];
+
+    int move_dir_x = 0;
+    int move_dir_y = 0;
+    if (action == ACTION_UP) move_dir_y = -1;
+    else if (action == ACTION_DOWN) move_dir_y = 1;
+    else if (action == ACTION_LEFT) move_dir_x = -1;
+    else if (action == ACTION_RIGHT) move_dir_x = 1;
+    else printf("Undefined action: %d", action);
+
+    int new_x = x + move_dir_x;
+    int new_y = y + move_dir_y;
+
+    if (new_x >= 0 && new_x < env->grid_size && new_y >= 0 && new_y < env->grid_size)
```

Comment:
Inverted conditional pattern here avoids nutty nesting

Reply:
Fixed (to be pushed).